lib/locale_threads.t

```diff
@@ -845,15 +845,17 @@ SKIP: {
     # and then a setlocale of the resulting one.  Discard locales which have
     # any unacceptable name
     if (${^O} eq "MSWin32" && $Config{'libc'} !~ /ucrt/) {
-        for my $i (0 .. $#locales) {
+        for (my $i = 0; $i < @locales; $i++) {
             my $locale_name = $locales[$i]->{locale_name};
             my $underlying_name = setlocale(&LC_CTYPE, $locale_name);
+            if (defined $underlying_name) {
 
-            # Defeat any attempt to skip the setlocale if the same as current,
-            # by switching to a locale very unlikey to be the current one.
-            setlocale($LC_ALL, "Albanian");
+                # Defeat any attempt to skip the setlocale if the same as current,
+                # by switching to a locale very unlikey to be the current one.
+                setlocale($LC_ALL, "Albanian");
 
-            next if setlocale(&LC_CTYPE, $underlying_name);
+                next if setlocale(&LC_CTYPE, $underlying_name);
+            }
 
             splice @locales, $i, 1;
```

Comment:
This still looks like it both splices and increments `$i` if `setlocale` fails, so it skips over some elements.

I think this loop should either be a `grep` ...
```perl
@locales = grep {
    my $locale_name = $locales[$i]->{locale_name};
    my $underlying_name = setlocale(&LC_CTYPE, $locale_name);

    # Defeat any attempt to skip the setlocale if the same as current,
    # by switching to a locale very unlikey to be the current one.
    setlocale($LC_ALL, "Albanian");

    defined($underlying_name) && setlocale(&LC_CTYPE, $underlying_name)
} @locales;
```
... or count backwards:
```perl
for my $i (reverse 0 .. $#locales) {
   my $locale_name = $locales[$i]->{locale_name};
    my $underlying_name = setlocale(&LC_CTYPE, $locale_name);

    # Defeat any attempt to skip the setlocale if the same as current,
    # by switching to a locale very unlikey to be the current one.
    setlocale($LC_ALL, "Albanian");

    if (!(defined($underlying_name) && setlocale(&LC_CTYPE, $underlying_name))) {
        splice @locales, $i, 1;
    }
}
```
... so `splice` does not affect the indices of not-yet-visited elements.

Reply:
I used your grep solution